src/lxml/objectify.pyx

```diff
@@ -1205,8 +1208,17 @@ cdef dict _PYTYPE_DICT = {}
 cdef dict _SCHEMA_TYPE_DICT = {}
 cdef list _TYPE_CHECKS = []
 
-cdef unicode _lower_bool(b):
-    return u"true" if b else u"false"
+cdef unicode _lower_bool(value):
+    return u"true" if value else u"false"
+
+cdef unicode _case_sensitive_float(value):
+    if math.isinf(value):
+        if value > 0:
+            return u"INF"
+        return u"-INF"
+    if math.isnan(value):
+        return u"NaN"
```

Comment:
Just a suggestion, but instead of `math`, you can import `isinf` and `isnan` directly as global `cdef` variables. Maybe rename them to `_float_is_inf` and `_float_is_nan`.

Reply:
Did what I believe you mean. Honestly I can't yet fully reason about putting `cdef object` along with `import` vs. omitting the `cdef`. I see there's an additional runtime `if` in the C code for the latter case, but don't know when to do what. If you find a minute, would you explain it to me?

Reply:
Sure. Defining a global variable with `cdef` makes it a static C variable. Otherwise, it's a Python variable that gets looked up in a dict. That's much slower. Also, `cdef` variables are not exposed to the module dict, i.e. not visible to users.